routing/pathfind.go

```diff
@@ -196,13 +196,14 @@ func (s sortableRoutes) Swap(i, j int) {
 
 // newRoute returns a fully valid route between the source and target that's
 // capable of supporting a payment of `amtToSend` after fees are fully
-// computed. If the route is too long, or the selected path cannot support the
-// fully payment including fees, then a non-nil error is returned.
+// computed. If the route is too long, the selected path cannot support the
+// fully payment including fees, or the path requires more than *maxFee fees,
+// then a non-nil error is returned.
 //
 // NOTE: The passed slice of ChannelHops MUST be sorted in forward order: from
 // the source to the target node of the path finding attempt.
 func newRoute(amtToSend lnwire.MilliSatoshi, pathEdges []*ChannelHop,
-	currentHeight uint32) (*Route, error) {
+	currentHeight uint32, maxFee *btcutil.Amount) (*Route, error) {
```

Comment:
I was hoping to use the interface created by the `oneof` as a parameter, but it is not exported. Instead, I take a pointer to the maximum fee in satoshis, or `nil` if there is no max fee. Let me know if there is a more idiomatic way to have an optional argument in Go, or if I'm totally missing another approach